test/e2e/providers/vsphere/vsphere.go

```diff
@@ -111,7 +111,7 @@ func (p *Provider) getWorkspaceFromExistingMachineSet() (*mapi.Workspace, error)
 // getNetwork returns the network that needs to be used in the MachineSet
 func getNetwork() string {
 	// Default network for dev environment
-	networkSegment := "dev-segment"
+	networkSegment := "devqe-segment-222"
```

Comment:
What is the point in making this change and immediately changing it in a follow up commit in the same PR? This is just unnecessary code churn.

Reply:
True, the follow-up commit was to highlight the additional (no longer needed) change.

Reply:
This should be pulled from the other machines in the cluster,  like the workspace is https://github.com/openshift/windows-machine-config-operator/pull/1792/files#diff-43febcc9af1775a8b8390c4d64b28993e558e2e58476773d1e1d4c78b5ff0b75R85

Otherwise if this changes again things will break

Reply:
Good point @sebsoto

Reply:
True. I looked at it while fixing this. I didn't feel it was worth the refactoring, plus losing the option of injecting the network name from env in CI (i.e. $LEASED_RESOURCE). 

Happy to implement the recommendation if that is the agreement. LMK.

Reply:
>True. I looked at it while fixing this. I didn't feel it was worth the refactoring, plus losing the option of injecting the network name from env in CI (i.e. $LEASED_RESOURCE).

What is the advantage of this option? @sebsoto's approach is the better long term fix given other workers will be using the correct  network.

Reply:
The current proposal is simpler and does not require refactoring in the e2e code, possibly adding more complexity (I know the `providerSpec` is there already 😄 ). 

I can't think of any implication on not using the `LEASED_RESOURCE`  env var right now, but maybe be some.

Reply:
>The current proposal is simpler and does not require refactoring in the e2e code, possibly adding more complexity (I know the providerSpec is there already 😄 ).

Simpler yes but it might cause us to revisit this again. In addition it forces developers to use a single segment. Yet another thing for them to keep in mind. The more I think about it, hard coding the segment feels like the wrong approach. Moreover I don't see the complexity in refactoring ` getWorkspaceFromExistingMachineSet()` into `getRequiredFieldsFromExistingMachineSet()` to also return the network segment.

Reply:
I have opened a separate PR with the proposed approach.

- https://github.com/openshift/windows-machine-config-operator/pull/1813